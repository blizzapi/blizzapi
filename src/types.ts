/** Common */

export type Uri = string;

/** Generic */

export enum ConstantKey {
  REGIONS = "REGIONS",
  LOCALES = "LOCALES",
  DEFAULT_LOCALES = "DEFAULT_LOCALES",
  REGION_API_HOSTS = "REGION_API_HOSTS",
  SC2_REALMS = "SC2_REALMS",
  OAUTH_AUTHORIZE_URIS = "OAUTH_AUTHORIZE_URIS",
  OAUTH_TOKEN_URIS = "OAUTH_TOKEN_URIS",
  OAUTH_CHECK_TOKEN_URIS = "OAUTH_CHECK_TOKEN_URIS",
}

/** Regions */

export enum RegionName {
  us = "us",
  eu = "eu",
  kr = "kr",
  tw = "tw",
  cn = "cn",
}

export enum RegionId {
  us = 1,
  eu = 2,
  kr = 3,
  tw = 3,
  cn = 5,
}

export enum RegionIdAsString {
  us = "1",
  eu = "2",
  kr = "3",
  tw = "3",
  cn = "5",
}

export type RegionIdKey = string;
export type RegionIdAsNumberOrString = RegionId | RegionIdAsString;
export type RegionIdArray = ReadonlyArray<RegionId>;
export type RegionNameArray = ReadonlyArray<RegionName>;

/** Locales */

export enum Locale {
  en_US = "en_US",
  es_MX = "es_MX",
  pt_BR = "pt_BR",
  en_GB = "en_GB",
  es_ES = "es_ES",
  fr_FR = "fr_FR",
  ru_RU = "ru_RU",
  de_DE = "de_DE",
  pt_PT = "pt_PT",
  it_IT = "it_IT",
  ko_KR = "ko_KR",
  zh_TW = "zh_TW",
  zh_CN = "zh_CN",
}

export type LocaleArray = ReadonlyArray<Locale>;

/** Default locales */

export enum DefaultLocaleIndex {
  us = 0,
  eu = 0,
  kr = 0,
  tw = 0,
  cn = 0,
}

export type DefaultLocaleIndexArray = ReadonlyArray<DefaultLocaleIndex>;

/** StarCraft II realms  */

export enum Sc2Realm {
  US = 1,
  LatAm = 2,
  Europe = 1,
  Russia = 2,
  Korea = 1,
  Taiwan = 2,
  China = 1,
}

export enum Sc2RealmAsString {
  US = "1",
  LatAm = "2",
  Europe = "1",
  Russia = "2",
  Korea = "1",
  Taiwan = "2",
  China = "1",
}

export type Sc2RealmArray = ReadonlyArray<Sc2Realm>;
export type Sc2RealmList = ReadonlyArray<Sc2Realm>;
export type Sc2RealmAsNumberOrString = Sc2Realm | Sc2RealmAsString;

/** API class types */

export type RegionIdOrName = RegionId | RegionName;
export type MaybeRegion = RegionIdOrName | undefined;
export type ClientId = string;
export type ClientSecret = string;
export type AccessToken = string | undefined;
export type Endpoint = string;

export enum HttpMethod {
  GET = "GET",
  POST = "POST",
}

export type ValidatorFunction = (endpoint: Endpoint) => boolean;

/* OAuth flow URIs */

export enum OAuthHost {
  us = "https://us.battle.net",
  eu = "https://eu.battle.net",
  kr = "https://apac.battle.net",
  tw = "https://apac.battle.net",
  cn = "https://www.battlenet.com.cn",
}

export enum RegionHost {
  us = "https://us.api.blizzard.com",
  eu = "https://eu.api.blizzard.com",
  kr = "https://kr.api.blizzard.com",
  tw = "https://tw.api.blizzard.com",
  cn = "https://gateway.battlenet.com.cn",
}

export enum OAuthEndpoint {
  authorize = "/oauth/authorize",
  token = "/oauth/token",
  checkToken = "/oauth/check_token?token=",
}

/** API headers */

export enum ApiHeaders {
  LastModified = "last-modified",
}

/** Errors */

export enum ErrorCode {
  NotAuthorized = 401,
}

export enum ErrorResponseMessage {
  InvalidToken = "invalid_token",
  AccessTokenInvalid = "access_token_invalid",
  AccessTokenExpired = "access_token_invalid",
}

export type InvalidAccessTokenError = {
  error: ErrorResponseMessage.AccessTokenInvalid;
};

export type ExpiredAccessTokenError = {
  error: ErrorResponseMessage.AccessTokenExpired;
};

export type ResponseError = InvalidAccessTokenError | ExpiredAccessTokenError;

/** Interfaces */

export interface AccessTokenOptions {
  validateAccessTokenOnEachQuery?: boolean;
  refreshExpiredAccessToken?: boolean;
  onAccessTokenExpired?: () => void;
  onAccessTokenRefresh?: (newAccessToken: string) => void;
}

export interface QueryOptions {
  timeout?: number;
  headers?: object | Headers;
  params?: object | URLSearchParams;
}

export interface BattleNetOptions {
  region: RegionIdOrName;
  clientId: ClientId;
  clientSecret: ClientSecret;
  accessToken?: AccessToken;
}

export interface InitOptions {
  region: string | number;
  clientId?: string;
  clientSecret?: string;
  accessToken?: string;
  validateAccessTokenOnEachQuery?: boolean;
  revalidateAccessTokenIfExpired?: boolean;
  onAccessTokenExpire?: () => void;
  onAccessTokenRevalidate?: () => void;
}

export interface BattleNetQueryOptions {
  region: RegionIdOrName;
  endpoint: string;
  clientId: string;
  clientSecret: string;
  accessToken: AccessToken;
  options: AccessTokenOptions & QueryOptions;
}

export type RegionIdProperties<Value> = {
  [key in RegionId]: Value;
};

export interface BlizzAPIOptions extends BattleNetOptions, AccessTokenOptions {}
