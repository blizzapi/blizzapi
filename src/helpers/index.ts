export * from "./bnet";
export * from "./fetch";
export * from "./oauth";
export * from "./validators";
