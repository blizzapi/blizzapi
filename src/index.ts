import BlizzAPI from './lib/classes/BlizzAPI';
export = BlizzAPI;