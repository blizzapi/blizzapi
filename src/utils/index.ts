export * from './localization';
export *  from './api';
export *  from './oauth';
