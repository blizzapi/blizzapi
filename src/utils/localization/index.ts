export * from "./regions";
export * from "./locales";
export * from "./defaultLocales";
export * from "./sc2Realms";
