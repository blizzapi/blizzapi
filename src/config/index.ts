/**
 * Class exposing configuration constants
 *
 * Via:
 *
 * https://develop.battle.net/documentation/guides/using-oauth
 * https://develop.battle.net/documentation/guides/regionality-partitions-and-localization
 */

import * as localization from 'config/localization';
import * as api from 'config/api';
import oauth from 'config/oauth';
import { ConfigObject } from 'types';

const config = {
  REGIONS: localization.regions,
  LOCALES: localization.locales,
  DEFAULT_LOCALES: localization.defaultLocales,
  SC2_REALMS: localization.sc2Realms,
  REGION_API_HOSTS: api.regionHosts,
  OAUTH_AUTHORIZE_URIS: oauth.authorizeUris,
  OAUTH_TOKEN_URIS: oauth.tokenUris,
  OAUTH_CHECK_TOKEN_URIS: oauth.checkTokenUris,
} as ConfigObject;

export default config;
