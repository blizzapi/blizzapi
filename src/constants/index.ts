/**
 * Class exposing configuration constants
 *
 * Via:
 *
 * https://develop.battle.net/documentation/guides/using-oauth
 * https://develop.battle.net/documentation/guides/regionality-partitions-and-localization
 */

import { ConstantKey } from "../types";
import * as localization from "./localization";
import * as api from "./api";
import * as oauth from "./oauth";

export const constants = {
  [ConstantKey.REGIONS]: localization.regions,
  [ConstantKey.LOCALES]: localization.locales,
  [ConstantKey.DEFAULT_LOCALES]: localization.defaultLocales,
  [ConstantKey.SC2_REALMS]: localization.sc2Realms,
  [ConstantKey.REGION_API_HOSTS]: api.regionHosts,
  [ConstantKey.OAUTH_AUTHORIZE_URIS]: oauth.authorizeUris,
  [ConstantKey.OAUTH_TOKEN_URIS]: oauth.tokenUris,
  [ConstantKey.OAUTH_CHECK_TOKEN_URIS]: oauth.checkTokenUris,
};
