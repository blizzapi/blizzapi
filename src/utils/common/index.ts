import constants from '../../constants';
import {
  validateRegionId,
  validateRegionName,
  getRegionIdByName,
} from '../localization/regions';
import {
  ConstantKey,
  RegionIdOrName,
  RegionIdAsNumberOrString,
  RegionName,
} from '../../types';

export const getConstantByRegionId = (
  regionId: RegionIdAsNumberOrString,
  constantKey: ConstantKey,
) => {
  const regionIdAsString = typeof regionId !== 'string'
    ? regionId
    : regionId.toString();
  const isRegionIdValid = validateRegionId(regionIdAsString);

  if (!isRegionIdValid) {
    throw new RangeError(
      `${regionIdAsString} is not a valid parameter for getConstantByRegionId(${regionId}, '${constantKey}')`,
    );
  }
  return constants[constantKey][regionIdAsString];
};

/**
 * Returns constant value for given region name from specified constant object
 *
 * @param regionName Region name as two-letter string (e.g. 'us')
 * @param constantKey Constant key
 * @return Uri as a string
 */
export const getConstantByRegionName = (regionName: RegionName, constantKey: ConstantKey) => {
  const isRegionNameValid = validateRegionName(regionName);

  if (!isRegionNameValid) {
    throw new RangeError(
      `${regionName} is not a valid parameter for getConstantByRegionName(${regionName}, '${constantKey}')`,
    );
  }

  const regionId = getRegionIdByName(regionName);

  return constants[constantKey][regionId];
};

export const getConstantByRegion = (regionIdOrName: RegionIdOrName, constantKey: ConstantKey) =>
  (validateRegionId(regionIdOrName)
    ? getConstantByRegionId(regionIdOrName, constantKey)
    : getConstantByRegionName(regionIdOrName.toString(), constantKey));
