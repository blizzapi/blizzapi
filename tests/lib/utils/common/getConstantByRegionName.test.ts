import { ConstantKey, RegionName } from "../../../../src/types";
import * as utils from "../../../../src/utils/common";

import regionNamesJson from "../../../__testData__/regionNames.json";
import wrongRegionNamesJson from "../../../__testData__/wrongRegionNames.json";
import constantKeysJson from "../../../__testData__/constantKeys.json";

const { getConstantByRegionName } = utils;

describe("getConstantByRegionName()", () => {
  it("should be defined", () => {
    expect(getConstantByRegionName).toBeDefined();
  });

  it('should be of type "function"', () => {
    expect(typeof getConstantByRegionName).toBe("function");
  });

  (regionNamesJson as ReadonlyArray<RegionName>).forEach((regionName) =>
    (constantKeysJson as ReadonlyArray<ConstantKey>).forEach((constantKey) => {
      it(`should return correct value for ${regionName} and ${constantKey} as valid parameters`, () => {
        expect(
          getConstantByRegionName(regionName, constantKey)
        ).toMatchSnapshot();
      });
    })
  );

  (wrongRegionNamesJson as ReadonlyArray<string>).forEach((wrongRegionName) =>
    (constantKeysJson as ReadonlyArray<ConstantKey>).forEach((constantKey) => {
      it(`should throw RangeError for ${wrongRegionName} and ${constantKey}`, () => {
        expect(() =>
          getConstantByRegionName(wrongRegionName as RegionName, constantKey)
        ).toThrow(RangeError);
      });
    })
  );
});
