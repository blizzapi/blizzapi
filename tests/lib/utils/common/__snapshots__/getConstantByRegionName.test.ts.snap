// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getConstantByRegionName() should return correct value for CN and OAUTH_AUTHORIZE_URIS as valid parameters 1`] = `"https://www.battlenet.com.cn/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for CN and OAUTH_CHECK_TOKEN_URIS as valid parameters 1`] = `"https://www.battlenet.com.cn/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for CN and OAUTH_TOKEN_URIS as valid parameters 1`] = `"https://www.battlenet.com.cn/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for CN and REGION_API_HOSTS as valid parameters 1`] = `"https://gateway.battlenet.com.cn"`;

exports[`getConstantByRegionName() should return correct value for Cn and OAUTH_AUTHORIZE_URIS as valid parameters 1`] = `"https://www.battlenet.com.cn/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for Cn and OAUTH_CHECK_TOKEN_URIS as valid parameters 1`] = `"https://www.battlenet.com.cn/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for Cn and OAUTH_TOKEN_URIS as valid parameters 1`] = `"https://www.battlenet.com.cn/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for Cn and REGION_API_HOSTS as valid parameters 1`] = `"https://gateway.battlenet.com.cn"`;

exports[`getConstantByRegionName() should return correct value for EU and OAUTH_AUTHORIZE_URIS as valid parameters 1`] = `"https://eu.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for EU and OAUTH_CHECK_TOKEN_URIS as valid parameters 1`] = `"https://eu.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for EU and OAUTH_TOKEN_URIS as valid parameters 1`] = `"https://eu.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for EU and REGION_API_HOSTS as valid parameters 1`] = `"https://eu.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for Eu and OAUTH_AUTHORIZE_URIS as valid parameters 1`] = `"https://eu.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for Eu and OAUTH_CHECK_TOKEN_URIS as valid parameters 1`] = `"https://eu.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for Eu and OAUTH_TOKEN_URIS as valid parameters 1`] = `"https://eu.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for Eu and REGION_API_HOSTS as valid parameters 1`] = `"https://eu.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for KR and OAUTH_AUTHORIZE_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for KR and OAUTH_CHECK_TOKEN_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for KR and OAUTH_TOKEN_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for KR and REGION_API_HOSTS as valid parameters 1`] = `
Array [
  "https://kr.api.blizzard.com",
  "https://tw.api.blizzard.com",
]
`;

exports[`getConstantByRegionName() should return correct value for Kr and OAUTH_AUTHORIZE_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for Kr and OAUTH_CHECK_TOKEN_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for Kr and OAUTH_TOKEN_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for Kr and REGION_API_HOSTS as valid parameters 1`] = `
Array [
  "https://kr.api.blizzard.com",
  "https://tw.api.blizzard.com",
]
`;

exports[`getConstantByRegionName() should return correct value for TW and OAUTH_AUTHORIZE_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for TW and OAUTH_CHECK_TOKEN_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for TW and OAUTH_TOKEN_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for TW and REGION_API_HOSTS as valid parameters 1`] = `
Array [
  "https://kr.api.blizzard.com",
  "https://tw.api.blizzard.com",
]
`;

exports[`getConstantByRegionName() should return correct value for Tw and OAUTH_AUTHORIZE_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for Tw and OAUTH_CHECK_TOKEN_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for Tw and OAUTH_TOKEN_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for Tw and REGION_API_HOSTS as valid parameters 1`] = `
Array [
  "https://kr.api.blizzard.com",
  "https://tw.api.blizzard.com",
]
`;

exports[`getConstantByRegionName() should return correct value for Us and OAUTH_AUTHORIZE_URIS as valid parameters 1`] = `"https://us.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for Us and OAUTH_CHECK_TOKEN_URIS as valid parameters 1`] = `"https://us.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for Us and OAUTH_TOKEN_URIS as valid parameters 1`] = `"https://us.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for Us and REGION_API_HOSTS as valid parameters 1`] = `"https://us.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for cN and OAUTH_AUTHORIZE_URIS as valid parameters 1`] = `"https://www.battlenet.com.cn/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for cN and OAUTH_CHECK_TOKEN_URIS as valid parameters 1`] = `"https://www.battlenet.com.cn/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for cN and OAUTH_TOKEN_URIS as valid parameters 1`] = `"https://www.battlenet.com.cn/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for cN and REGION_API_HOSTS as valid parameters 1`] = `"https://gateway.battlenet.com.cn"`;

exports[`getConstantByRegionName() should return correct value for cn and OAUTH_AUTHORIZE_URIS as valid parameters 1`] = `"https://www.battlenet.com.cn/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for cn and OAUTH_CHECK_TOKEN_URIS as valid parameters 1`] = `"https://www.battlenet.com.cn/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for cn and OAUTH_TOKEN_URIS as valid parameters 1`] = `"https://www.battlenet.com.cn/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for cn and REGION_API_HOSTS as valid parameters 1`] = `"https://gateway.battlenet.com.cn"`;

exports[`getConstantByRegionName() should return correct value for eU and OAUTH_AUTHORIZE_URIS as valid parameters 1`] = `"https://eu.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for eU and OAUTH_CHECK_TOKEN_URIS as valid parameters 1`] = `"https://eu.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for eU and OAUTH_TOKEN_URIS as valid parameters 1`] = `"https://eu.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for eU and REGION_API_HOSTS as valid parameters 1`] = `"https://eu.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for eu and OAUTH_AUTHORIZE_URIS as valid parameters 1`] = `"https://eu.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for eu and OAUTH_CHECK_TOKEN_URIS as valid parameters 1`] = `"https://eu.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for eu and OAUTH_TOKEN_URIS as valid parameters 1`] = `"https://eu.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for eu and REGION_API_HOSTS as valid parameters 1`] = `"https://eu.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for kR and OAUTH_AUTHORIZE_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for kR and OAUTH_CHECK_TOKEN_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for kR and OAUTH_TOKEN_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for kR and REGION_API_HOSTS as valid parameters 1`] = `
Array [
  "https://kr.api.blizzard.com",
  "https://tw.api.blizzard.com",
]
`;

exports[`getConstantByRegionName() should return correct value for kr and OAUTH_AUTHORIZE_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for kr and OAUTH_CHECK_TOKEN_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for kr and OAUTH_TOKEN_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for kr and REGION_API_HOSTS as valid parameters 1`] = `
Array [
  "https://kr.api.blizzard.com",
  "https://tw.api.blizzard.com",
]
`;

exports[`getConstantByRegionName() should return correct value for tW and OAUTH_AUTHORIZE_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for tW and OAUTH_CHECK_TOKEN_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for tW and OAUTH_TOKEN_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for tW and REGION_API_HOSTS as valid parameters 1`] = `
Array [
  "https://kr.api.blizzard.com",
  "https://tw.api.blizzard.com",
]
`;

exports[`getConstantByRegionName() should return correct value for tw and OAUTH_AUTHORIZE_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for tw and OAUTH_CHECK_TOKEN_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for tw and OAUTH_TOKEN_URIS as valid parameters 1`] = `"https://apac.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for tw and REGION_API_HOSTS as valid parameters 1`] = `
Array [
  "https://kr.api.blizzard.com",
  "https://tw.api.blizzard.com",
]
`;

exports[`getConstantByRegionName() should return correct value for uS and OAUTH_AUTHORIZE_URIS as valid parameters 1`] = `"https://us.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for uS and OAUTH_CHECK_TOKEN_URIS as valid parameters 1`] = `"https://us.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for uS and OAUTH_TOKEN_URIS as valid parameters 1`] = `"https://us.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for uS and REGION_API_HOSTS as valid parameters 1`] = `"https://us.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for us and OAUTH_AUTHORIZE_URIS as valid parameters 1`] = `"https://us.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for us and OAUTH_AUTHORIZE_URIS as valid parameters 2`] = `"https://us.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for us and OAUTH_CHECK_TOKEN_URIS as valid parameters 1`] = `"https://us.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for us and OAUTH_CHECK_TOKEN_URIS as valid parameters 2`] = `"https://us.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for us and OAUTH_TOKEN_URIS as valid parameters 1`] = `"https://us.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for us and OAUTH_TOKEN_URIS as valid parameters 2`] = `"https://us.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for us and REGION_API_HOSTS as valid parameters 1`] = `"https://us.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for us and REGION_API_HOSTS as valid parameters 2`] = `"https://us.api.blizzard.com"`;
